web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/CreateDraftCertificateFromCS.java

```diff
@@ -0,0 +1,115 @@
+import se.inera.intyg.webcert.web.csintegration.integration.CSIntegrationRequestFactory;
+import se.inera.intyg.webcert.web.csintegration.integration.CSIntegrationService;
+import se.inera.intyg.webcert.web.csintegration.util.PDLLogService;
+import se.inera.intyg.webcert.web.integration.interactions.createdraftcertificate.v3.CreateDraftCertificate;
+import se.inera.intyg.webcert.web.integration.interactions.createdraftcertificate.v3.CreateDraftCertificateResponseFactory;
+import se.inera.intyg.webcert.web.integration.registry.IntegreradeEnheterRegistry;
+import se.inera.intyg.webcert.web.integration.registry.dto.IntegreradEnhetEntry;
+import se.inera.intyg.webcert.web.service.monitoring.MonitoringLogService;
+import se.inera.intyg.webcert.web.service.patient.PatientDetailsResolver;
+import se.riv.clinicalprocess.healthcond.certificate.createdraftcertificateresponder.v3.CreateDraftCertificateResponseType;
+import se.riv.clinicalprocess.healthcond.certificate.createdraftcertificateresponder.v3.Intyg;
+import se.riv.clinicalprocess.healthcond.certificate.v3.ErrorIdType;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CreateDraftCertificateFromCS implements CreateDraftCertificate {
+
+    private final PatientDetailsResolver patientDetailsResolver;
+    private final CSIntegrationService csIntegrationService;
+    private final CSIntegrationRequestFactory csIntegrationRequestFactory;
+    private final IntegreradeEnheterRegistry integreradeEnheterRegistry;
+    private final PDLLogService pdlLogService;
+    private final MonitoringLogService monitoringLogService;
+
+    @Override
+    public CreateDraftCertificateResponseType create(Intyg certificate, IntygUser user) {
+        final var applicationError = validatePUIntegration(certificate);
+        if (applicationError.isPresent()) {
+            return applicationError.get();
+        }
+
+        final var modelId = csIntegrationService.certificateTypeExists(certificate.getTypAvIntyg().getCode());
+        if (modelId.isEmpty()) {
+            log.debug("Certificate type '{}' does not exist in certificate service", certificate.getTypAvIntyg().getCode());
+            return null;
+        }
+
+        integreradeEnheterRegistry.putIntegreradEnhet(getIntegreradEnhetEntry(user), false, true);
+
+        try {
+            final var createdCertificate = csIntegrationService.createCertificate(
+                csIntegrationRequestFactory.createDraftCertificateRequest(modelId.get(), certificate, user)
+            );
+
+            pdlLogService.logCreatedWithIntygUser(createdCertificate, user);
+            monitoringLogService.logUtkastCreated(
+                createdCertificate.getMetadata().getId(),
+                createdCertificate.getMetadata().getType(),
+                createdCertificate.getMetadata().getUnit().getUnitId(),
+                createdCertificate.getMetadata().getIssuedBy().getPersonId(),
+                0
```

Comment:
Suggest to create a constant for the "0". Call it something like NO_PREFILL_ELEMENTS.